lib/Installation/Partitioner/LibstorageNG/v4_3/GuidedSetupController.pm

```diff
@@ -0,0 +1,84 @@
+# SUSE's openQA tests
+#
+# Copyright © 2021 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved. This file is offered as-is,
+# without any warranty.
+
+# Summary: The class introduces business actions for Guided Setup of
+# Libstorage-NG Partitioner using REST API.
+
+# Maintainer: QE YaST <qa-sle-yast@suse.de>
+
+package Installation::Partitioner::LibstorageNG::v4_3::GuidedSetupController;
+
+use parent 'Installation::Partitioner::LibstorageNG::GuidedSetupController';
+use strict;
+use warnings;
+
+use Installation::Partitioner::LibstorageNG::v4_3::SelectDisksToUsePage;
+use Installation::Partitioner::LibstorageNG::v4_3::SelectDisksPartitionsPage;
+
+=head1 PARTITION_SETUP
+
+=head2 SYNOPSIS
+
+The class introduces business actions for Guided Setup of Libstorage-NG
+Partitioner using REST API.
+
+=cut
+
+
+
+sub new {
+    my ($class, $args) = @_;
+    my $self = bless {}, $class;
+    return $self->init($args);
+}
+
+sub init {
+    my ($self, $args) = @_;
+    $self->SUPER::init($args);
+    $self->{SelectDisksToUsePage}      = Installation::Partitioner::LibstorageNG::v4_3::SelectDisksToUsePage->new({app => YuiRestClient::get_app()});
+    $self->{SelectDisksPartitionsPage} = Installation::Partitioner::LibstorageNG::v4_3::SelectDisksPartitionsPage->new({app => YuiRestClient::get_app()});
+    return $self;
+}
+
+sub get_select_disks_partitions_page {
+    my ($self) = @_;
+    return $self->{SelectDisksPartitionsPage};
+}
+
+sub get_select_disks_to_use_page {
+    my ($self) = @_;
+    return $self->{SelectDisksToUsePage};
+}
+
+=head2 guided_setup
+
+ guided_setup($self, %args);
+
+Method setups partitioning using guided setup using provided configuration details.
+Following keys can be used:
+C<$args{disks}> - Define list of disks to be used (in case of multidisk setup)
+C<$args{existing_partitions}> - Process existing partitions, in case those do exist
+on the selected disks
+
+=cut
+
+sub guided_setup {
+    my ($self, %args) = @_;
+    $self->get_suggested_partitioning_page()->press_guided_setup_button();
+    if (my @disks = @{$args{disks}}) {
+        $self->get_select_disks_to_use_page()->select_hard_disks(@disks);
+    }
+    if ($args{existing_partitions}) {
+        $self->get_select_disks_partitions_page()->skip_handling_partitions();
```

Comment:
what about next if the key is not defined putting the is_shown every time we get the page? We could just have some setup method which is partially implemented which does next at the end, and when we need more we just put there some code. Do you like more to use skip explicitly for some reason, I'm facing similar things, so I'm not sure. What I remember we agreed a time ago was to have different methods if there is different flow of screens but it is not this case.

Reply:
in fact, at least the first one is the case, where the screen is optional depending of the number or disk, the other one I don't know, I didn't see it in the verifications, so perhaps two different methods make it more clear, but there is not big different though.

Reply:
This is the way we have it currently in older guided setup, in general we don't do anything on the screen with existing partitions. I've realized that I didn't have to implement this either, as `select_disk` is not executed anywhere where we expect existing partitions. Which is not the case in lvm tests where we boot into image created by `cryptlvm` scenario. There we need this functionality. As for naming, I will spend more time on it, that's why PR is WIP. But `skip_handling_partitions` is the name we already have in the previous implementation.